modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/source/PollingSourceNotificationsTestCase.java

```diff
@@ -0,0 +1,252 @@
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.module.extension.source;
+
+import static java.lang.String.valueOf;
+import static java.util.Collections.singletonMap;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mule.runtime.api.notification.PipelineMessageNotification.PROCESS_COMPLETE;
+import static org.mule.runtime.api.notification.PollingSourceItemNotification.ITEM_REJECTED_IDEMPOTENCY;
+import static org.mule.runtime.api.notification.PollingSourceItemNotification.ITEM_REJECTED_LIMIT;
+import static org.mule.runtime.api.notification.PollingSourceItemNotification.ITEM_REJECTED_WATERMARK;
+import static org.mule.runtime.api.notification.PollingSourceNotification.POLL_FAILURE;
+import static org.mule.runtime.api.notification.PollingSourceNotification.POLL_SUCCESS;
+import static org.mule.runtime.module.extension.internal.ExtensionProperties.ENABLE_POLLING_SOURCE_LIMIT_PARAMETER;
+import static org.mule.test.allure.AllureConstants.SourcesFeature.SOURCES;
+import static org.mule.test.allure.AllureConstants.SourcesFeature.SourcesStories.POLLING;
+import static org.mule.test.petstore.extension.PetAdoptionSource.ALL_PETS;
+import static org.mule.test.petstore.extension.WatermarkingPetAdoptionSource.resetSource;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.lifecycle.Startable;
+import org.mule.runtime.api.notification.Notification;
+import org.mule.runtime.api.notification.PipelineMessageNotification;
+import org.mule.runtime.api.notification.PipelineMessageNotificationListener;
+import org.mule.runtime.api.notification.PollingSourceItemNotification;
+import org.mule.runtime.api.notification.PollingSourceItemNotificationListener;
+import org.mule.runtime.api.notification.PollingSourceNotification;
+import org.mule.runtime.api.notification.PollingSourceNotificationListener;
+import org.mule.runtime.api.util.concurrent.Latch;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.api.processor.Processor;
+import org.mule.tck.junit4.rule.SystemProperty;
+import org.mule.test.module.extension.AbstractExtensionFunctionalTestCase;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.junit.Rule;
+import org.junit.Test;
+
+@Feature(SOURCES)
+@Story(POLLING)
+public class PollingSourceNotificationsTestCase extends AbstractExtensionFunctionalTestCase {
+
+  private static final List<CoreEvent> ADOPTION_EVENTS = new LinkedList<>();
+
+  @Rule
+  public SystemProperty emitNotifications = new SystemProperty("mule.emit.polling.source.notifications", "true");
+
+  @Override
+  protected boolean mustRegenerateExtensionModels() {
+    return true;
```

Comment:
why?

Reply:
this is needed to access the poll item limit feature we added

Reply:
not the answer i was looking for. Why do you need to regenerate the entire extension model just for this?

Reply:
still unresolved

Reply:
I need to use the item limit feature and I copied this code from https://github.com/mulesoft/mule/blob/44faefefef1d33316342f0122e64aa22f3680566/modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/source/PollingSourceLimitingTestCase.java#L68
Look into the PR that introduced the feature to know why it was needed:
https://github.com/mulesoft/mule/commit/83d46ead6204e4e0849d4f9fb62a089332d134ef

Reply:
why do you need that feature to test this? You should be able to do this with a simple CountDownLatch